DDF-1633 Fix Admin UI source tab bugs

harrison-tarr opened the pull request:
Could create duplicate service configurations of the same type under
the same Source name
Could not create multiple service configurations of different types
under the same Source name

DDF-1633 Add "delete source" check

Deleting each service config was getting annoying during debugging
so I added the ability to delete all configs. This should be useful
to admins irl too.

DDF-1633 Allow editing a disabled source

ConfigurationAdmin would only match a metatype if it was enabled
now it will match in both case and allow updating

DDF-1633 Fix delete source

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/codice/ddf/328)
<!-- Reviewable:end -->

harrison-tarr commented:
@vinamartin @djblue @andrewkfiedler 
To hero:
build normally and then ensure you can do the following:
-create source/service
	-should be created as disabled, unless first in that source name
-not create source with same name & type as existing
-edit source/service
	-editing source/service should edit not add
-change current config with dropdown
-change edit config with drop down in modal and edit disabled config w/o enabling
	-edits should be saved
-delete services
	-including by source

let me know if you need more explanation for any of these

vinamartin commented on catalog/admin/module/catalog-admin-module-sources/src/main/webapp/templates/deleteSource.handlebars:
```diff
@@ -11,7 +11,10 @@
  **/
  --}}
-<div class="delete-source-name">{{name}}</div>
+<div class="delete-source-name"><span style="margin-right: 2em">{{name}}</span>
```
It'd be nice to make this look slightly nicer, but I'm not sure what the desired look would be.

harrison-tarr replied:
This currently looks like: ![screen shot 2015-11-10 at 2 29 20 pm](https://cloud.githubusercontent.com/assets/14062223/11076412/dd87f5f6-87b7-11e5-89f9-8a385890a305.png)

I also had a version where the "Delete Source" checkbox was in the same column as the other configurations under that source.

Does anyone have thoughts or preferences regarding the look?

vinamartin replied:
I would try moving `Delete Source` to the end of the list and see how that looks instead

harrison-tarr replied:
New iteration: 
![screen shot 2015-11-11 at 9 41 55 am](https://cloud.githubusercontent.com/assets/14062223/11096752/e41fcf00-8858-11e5-98cc-d0873cd5e04b.png)

The source checkbox (next to the source name) will select/de-select all services underneath. If one service is deselected, it will also deselect the source checkbox. If all services are manually checked, it will also check the source checkbox.

vinamartin replied:
This is good! :+1:

clockard commented:
Built successfully with all tests passing. Manual checks/tests passed.
:+1:

andrewkfiedler commented:
Review status: 0 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

<sup>**[catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js, line 151 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/328#-K2sQl9NpJ9HEenCHbBU)** ([raw file](https://github.com/codice/ddf/blob/ed85414ddd1cd22285ba1d414d53abeafc9ff79a/catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js#L151)):</sup>
Any reason to not return the ajax call directly?

Also, what happens if there's no pid?  Do we never resolve the deferred?
<!-- Reviewable comment -K2sQl9NpJ9HEenCHbBV:-1516777325 -->

---

<sup>**[catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js, line 173 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/328#-K2sRBNKA6BuUllUBe4N)** ([raw file](https://github.com/codice/ddf/blob/ed85414ddd1cd22285ba1d414d53abeafc9ff79a/catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js#L173)):</sup>
You can rewrite this whole nested ```if``` statement block to be more concise and readable.  I'll come by and show you how.  
<!-- Reviewable comment -K2sRBNKA6BuUllUBe4O:-460256065 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/328)
<!-- Sent from Reviewable.io -->

andrewkfiedler commented:
Review status: 0 of 6 files reviewed at latest revision, 4 unresolved discussions.

---

<sup>**[catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Source.js, line 57 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/328#-K2sSnScBuJELkUWKHng)** ([raw file](https://github.com/codice/ddf/blob/ed85414ddd1cd22285ba1d414d53abeafc9ff79a/catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Source.js#L57)):</sup>
Just get the configuration and check it directly.
<!-- Reviewable comment -K2sSnScBuJELkUWKHnh:-490186122 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/328)
<!-- Sent from Reviewable.io -->

harrison-tarr commented:
Review status: 0 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

<sup>**[catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js, line 151 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/328#-K2sQl9NpJ9HEenCHbBU)** ([raw file](https://github.com/codice/ddf/blob/ed85414ddd1cd22285ba1d414d53abeafc9ff79a/catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js#L151)):</sup>
fixed.
<!-- Reviewable comment -K2sfPE0wLvS1wNbjye2:-1620474753 -->

---

<sup>**[catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Source.js, line 57 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/328#-K2sSnScBuJELkUWKHng)** ([raw file](https://github.com/codice/ddf/blob/ed85414ddd1cd22285ba1d414d53abeafc9ff79a/catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Source.js#L57)):</sup>
fixed.
<!-- Reviewable comment -K2sfRHmZq5HOgivyWMD:-1620474753 -->

---

<sup>**[catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js, line 173 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/328#-K2sRBNKA6BuUllUBe4N)** ([raw file](https://github.com/codice/ddf/blob/ed85414ddd1cd22285ba1d414d53abeafc9ff79a/catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js#L173)):</sup>
fixed.
<!-- Reviewable comment -K2sfTH62feCJQVCEWyK:-1620474753 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/328)
<!-- Sent from Reviewable.io -->

andrewkfiedler commented:
Reviewed 1 of 6 files at r1, 1 of 5 files at r2, 4 of 4 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/328)
<!-- Sent from Reviewable.io -->

andrewkfiedler commented:
:+1:
